### PR TITLE
New Packets (Freeze Animation, AddUnitFOW, another FloatingText, Input Lock) + PacketEnum sorted + .rainbow

### PR DESCRIPTION
**Pull request by Maufeat**

### CODE REVIEW

**Comment by molenzwiebel** [comment]
Your packet still reads `AddUnitFOG` instead of `AddUnitFOW`

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                    //sending the actual Thread of execution to sleep X milliseconds
+                    float a = 0.5f;
+                    byte[] rainBow = new byte[4];
+                    Random random = new Random();
+                    random.NextBytes(rainBow);
+                    Thread.Sleep(500);
+                    SetScreenTint s1 = new SetScreenTint(me, false, 0.0f, 0, 0, 0, 1f);
+                    owner.GetGame().PacketHandlerManager.broadcastPacket(s1, Core.Logic.PacketHandlers.Channel.CHL_S2C);
+                    SetScreenTint sst = new SetScreenTint(me, true, 0.6f, rainBow[1], rainBow[2], rainBow[3], a);
+                    owner.GetGame().PacketNotifier.notifyDebugMessage("R: " + rainBow[1] + " G: " + rainBow[2] + " B: " + rainBow[3]);
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: use `string.Format("R: {0} G: {1} B: {2}", rainBow[1], rainBow[2], rainBow[3]);` instead, and format it to be < 120 characters in the line if necessary.

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        public void RAINBOW()
+        {
+            while (true)
+            {
+                try
+                {
+                    //sending the actual Thread of execution to sleep X milliseconds
+                    float a = 0.5f;
+                    byte[] rainBow = new byte[4];
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: This should just be called `rainbow`

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        private static ChatboxManager owner;
+
+        public override void Execute(Peer peer, bool hasReceivedArguments, string arguments = "")
+        {
+            me = _owner.GetGame().GetPeerInfo(peer).GetChampion();
+            owner = _owner;
+            Task.Run(() => RAINBOW());
+        }
+
+        public void RAINBOW()
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: Why the full caps name?

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            while (true)
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: Please no

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                    owner.GetGame().PacketNotifier.notifyDebugMessage("R: " + rainBow[1] + " G: " + rainBow[2] + " B: " + rainBow[3]);
+                    owner.GetGame().PacketHandlerManager.broadcastPacket(sst, Core.Logic.PacketHandlers.Channel.CHL_S2C);
+                }
+                catch (Exception e)
+                {
+                }
+            }
+        }
+    }
+}
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: Missing newline

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public void RAINBOW()
+        {
+            while (true)
+            {
+                try
+                {
+                    //sending the actual Thread of execution to sleep X milliseconds
+                    float a = 0.5f;
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: Why do you re-declare this variable every time the loop occurs? Shouldn't this be moved outside the loop?

**Comment by MythicManiac** [comment]
Address my comments and I'll merge. So either make changes or clarify why you keep them as is.

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            while (run)
+            {
+                try
+                {
+                    byte[] rainbow = new byte[4];
+                    new Random().NextBytes(rainbow);
+                    Thread.Sleep(delay);
+                    SetScreenTint tintOff = new SetScreenTint(me, false, 0.0f, 0, 0, 0, 1f);
+                    owner.GetGame().PacketHandlerManager.broadcastPacket(tintOff, Core.Logic.PacketHandlers.Channel.CHL_S2C);
+                    if (run)
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: I don't think this check is necessary

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            }
+            else
+            {
+                run = false;
+                SetScreenTint tintOff = new SetScreenTint(me, false, 0.0f, 0, 0, 0, 1f);
+                owner.GetGame().PacketHandlerManager.broadcastPacket(tintOff, Core.Logic.PacketHandlers.Channel.CHL_S2C);
+            }
+        }
+
+        public void rainbow()
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: `public void Rainbow()` ?

File: GameServer/Logic/Chatbox/Commands/RainbowCommand.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                float.TryParse(split[1], out a);
+            }
+
+            if(split.Length > 2)
+            {
+                speed = float.Parse(split[2]);
+                delay = (int)(speed * 1000);
+            }
+
+            if (!run)
```
Comment (MythicManiac) [inline GameServer/Logic/Chatbox/Commands/RainbowCommand.cs]: This section could be shortened to

```CS
run = !run
if(run)
{
    Task.Run(() => TaskRainbow());
}
```

**Comment by MythicManiac** [comment]
This should be good enough for now—merging after tests pass

**Pull request merged**